src/engine/contactwriter.cpp

```diff
@@ -2530,6 +2546,64 @@ static ContactWriter::DetailList allSingularDetails()
     return details;
 }
 
+static QContactDetail supportedDetailInstance(QContactDetail::DetailType type)
+{
+    switch (type) {
```

Comment:
Thanks, updated.  Also I changed the test at line 4734 of contactwriter.cpp to check for differing types rather than just checking for undefined type.